api/src/main/scala/hmda/api/http/InstitutionsHttpApi.scala

```diff
@@ -41,39 +41,43 @@ trait InstitutionsHttpApi extends InstitutionProtocol with ApiErrorProtocol with
   val institutionsPath =
     path("institutions") {
       val path = "institutions"
-      val institutionsActor = system.actorSelection("/user/institutions")
-      timedGet {
-        val fInstitutions = (institutionsActor ? GetState).mapTo[Set[Institution]]
-        onComplete(fInstitutions) {
-          case Success(institutions) =>
-            val wrappedInstitutions = institutions.map(inst => InstitutionWrapper(inst.id, inst.name, inst.status))
-            complete(ToResponseMarshallable(Institutions(wrappedInstitutions)))
-          case Failure(error) => completeWithInternalError(path, error)
+      hmdaAuthorize {
+        val institutionsActor = system.actorSelection("/user/institutions")
+        timedGet {
+          val fInstitutions = (institutionsActor ? GetState).mapTo[Set[Institution]]
+          onComplete(fInstitutions) {
+            case Success(institutions) =>
+              val wrappedInstitutions = institutions.map(inst => InstitutionWrapper(inst.id, inst.name, inst.status))
+              complete(ToResponseMarshallable(Institutions(wrappedInstitutions)))
+            case Failure(error) => completeWithInternalError(path, error)
+          }
         }
       }
     }
 
   val institutionByIdPath =
     path("institutions" / Segment) { institutionId =>
       val path = s"institutions/$institutionId"
-      extractExecutionContext { executor =>
-        val institutionsActor = system.actorSelection("/user/institutions")
-        val filingsActor = system.actorOf(FilingPersistence.props(institutionId))
-        timedGet {
-          implicit val ec: ExecutionContext = executor
-          val fInstitutionDetails = institutionDetails(institutionId, institutionsActor, filingsActor)
-          onComplete(fInstitutionDetails) {
-            case Success(institutionDetails) =>
-              filingsActor ! Shutdown
-              if (institutionDetails.institution.name != "")
-                complete(ToResponseMarshallable(institutionDetails))
-              else {
-                val errorResponse = ErrorResponse(404, s"Institution $institutionId not found", path)
-                complete(ToResponseMarshallable(StatusCodes.NotFound -> errorResponse))
-              }
-            case Failure(error) =>
-              filingsActor ! Shutdown
-              completeWithInternalError(path, error)
+      hmdaAuthorize {
+        extractExecutionContext { executor =>
+          val institutionsActor = system.actorSelection("/user/institutions")
+          val filingsActor = system.actorOf(FilingPersistence.props(institutionId))
```

Comment:
Yep, that's part of a separate issue: https://github.com/cfpb/hmda-platform/issues/504. It was too much to implement all in one go.